src/network/transaction_request/mod.rs

```diff
@@ -46,6 +51,17 @@ impl TransactionRequest {
         self
     }
 
+    pub fn set_factory_deps(&mut self, factory_deps: Vec<Bytes>) {
```

Comment:
Could you explain what use cases do you see for this method that are not covered by `zksync_deploy` method?

Reply:
This offers a consistent API to the `factory_deps` field as a user. An example use case is during `forge create` where the factory deps are populated via cli args.

https://github.com/matter-labs/foundry-zksync/pull/608/files#diff-2cfb15a508e13ae934f28a7452a64a1570768cd367a66d052d4129e7f37a6868R592